src/NetworkName.php

```diff
@@ -475,25 +475,39 @@ public static function showFormForNetworkPort($networkPortID)
         );
         echo "</td>\n";
 
-        echo "</tr><tr class='tab_bg_1'>\n";
-
-        echo "<td>" . IPAddress::getTypeName(Session::getPluralNumber());
-        IPAddress::showAddChildButtonForItemForm($name, 'NetworkName__ipaddresses');
-        echo "</td>";
-        echo "<td>";
-        IPAddress::showChildsForItemForm($name, 'NetworkName__ipaddresses');
-        echo "</td>";
-
-       // MoYo : really need to display it here ?
-       // make confure because not updatable
-       // echo "<td>".IPNetwork::getTypeName(Session::getPluralNumber())."&nbsp;";
-       // Html::showToolTip(__('IP network is not included in the database. However, you can see current available networks.'));
-       // echo "</td><td>";
-       // IPNetwork::showIPNetworkProperties($name->getEntityID());
-       // echo "</td>\n";
-        echo "<td colspan='2'>&nbsp;</td>";
-
-        echo "</tr>\n";
+        echo "</tr>";
+
+        if ($name->isNewItem()) {
+            if (!$name->canCreate()) {
+                $canedit = false;
+            }
+            $canedit = $name->canUpdate();
+        } else {
+            if (!$name->can($name->getID(), READ)) {
+                $canedit = false;
+            }
+            $canedit = $name->can($name->getID(), UPDATE);
+        }
+
+        if ($canedit) {
+            echo "<tr class='tab_bg_1'>\n";
+            echo "<td>" . IPAddress::getTypeName(Session::getPluralNumber());
+            IPAddress::showAddChildButtonForItemForm($name, 'NetworkName__ipaddresses', $canedit);
+            echo "</td>";
+            echo "<td>";
+            IPAddress::showChildsForItemForm($name, 'NetworkName__ipaddresses', $canedit);
+            echo "</td>";
+
+           // MoYo : really need to display it here ?
```

Comment:
This comment has 10 years, I guess you can profit to delete this part

Reply:
agree with that